drivers: arm_cmsdk/arm: Convert drivers to new DT_INST macros

galak opened the pull request:
Convert older DT_INST_ macro use in arm_cmsdk/arm drivers to the new
include/devicetree.h DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: arm_cmsdk/arm: Convert drivers to new DT_INST macros

Convert older DT_INST_ macro use in arm_cmsdk/arm drivers to the new
include/devicetree.h DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
soc: arm: qemu_cortex_a53: Convert to new DT_INST macros

Convert older DT_INST_ macro use the new include/devicetree.h
DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

pabigot commented on drivers/counter/timer_tmr_cmsdk_apb.c:
```diff
@@ -171,38 +173,38 @@ static const struct tmr_cmsdk_apb_cfg tmr_cmsdk_apb_cfg_0 = {
 static void timer_cmsdk_apb_config_0(struct device *dev)
 {
-	IRQ_CONNECT(DT_INST_0_ARM_CMSDK_TIMER_IRQ_0, DT_INST_0_ARM_CMSDK_TIMER_IRQ_0_PRIORITY,
+	IRQ_CONNECT(DT_INST_IRQN(0), DT_INST_IRQN(0),
```
This can't be right; the second argument should be a priority.  Would explain the shippable failure.

Same issue in other instances in this fille; some other files get it right.

galak replied:
Good catch, now fixed.

stephanosio commented on drivers/ipm/ipm_mhu.c:
```diff
@@ -195,18 +197,18 @@ DEVICE_AND_API_INIT(mhu_0,
-	IRQ_CONNECT(DT_INST_0_ARM_MHU_IRQ_0,
-			DT_INST_0_ARM_MHU_IRQ_0,
+	IRQ_CONNECT(DT_INST_IRQN(0),
+			DT_INST_IRQN(0),
```
While the conversion done here is correct, this does not look sane to me ...

galak replied:
Agreed, seems like that should really be looking for `priority`.  Will fix it up.

galak committed:
drivers: ipm: ipm_mhu: Fix typo bug in IRQ_CONNECT

The second paramater to IRQ_CONNECT for ipm_mhu_irq_config_func_0 should
be passing the priority, instead it passed the IRQ number.  Fix this to
pass priority (which matches ipm_mhu_irq_config_func_1).

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>
